packages/webpack/src/plugin.js

```diff
@@ -221,11 +225,38 @@ class LavaMoatPlugin {
           JAVASCRIPT_MODULE_TYPE_ESM,
         ]
 
+        // Webpack has a concept of ignored modules
+        // When a module is ignored a carveout is necessary in policy enforcement for it because the ID that webpack creates for it is not exactly helpful.
+        // example outcome in the bundle: `const nodeCrypto = __webpack_require__(/*! crypto */ "?0b7d");`
+        // Sadly, even treeshaking doesn't eliminate that module. It's left there and failing to work when reached by runtime policy enforcement.
+        // Below is the most reliable way I've found to date to identify ignored modules.
+        /**
+         * @param {import('webpack').Module} m
+         * @returns {boolean}
+         */
+        const isIgnoredModule = (m) => {
+          if (
+            m.type === JAVASCRIPT_MODULE_TYPE_DYNAMIC &&
+            // @ts-expect-error BAD TYPES
+            m.identifierStr?.startsWith('ignored')
+          ) {
+            return true
+          }
+          // This function must be very narrow, because it's gatekeeping policy generation too
+          return false
+        }
+
         /**
          * @param {import('webpack').Module} m
-         * @returns {m is import('webpack').NormalModule}
+         * @param {string} moduleClass
+         * @returns {m is import('webpack').NormalModule} // TODO: this is not
+         *   true anymore, but there's no superclass of all reasonable module
+         *   types
```

Comment:
If this isn't doing anything useful, just return `boolean`

Reply:
It is useful as it helps with what fields are available. Module type barely has any fields. Normal module is the most common one, but not the only one we want to let pass through this function. But there's no other type that all modules with actual code in them inherit from and merging the ones we know is not future proof.

This makes types helpful despite being not exactly true.

Reply:
If there is no appropriate enough type available already, better to make one that captures the behavior of relevance to us, rather than reaching for the closest available one? Just capturing the fields that you think are useful now, I think no need for it to be more complete or exact than `NormalModule`.

Reply:
(IMO: Overly vague type signatures for return values is generally fine but incomplete ones are not. Inverse for input parameters)

Reply:
okay, so if the type guard is doing actual work--good, just keep it that way.  though I'd probably lean more in the direction of what @legobeat said; create a new typedef (intersecting Module, if appropriate) containing the properties we care about, and use that.  not a big enough deal to block over though

Reply:
We need to talk about this synchronously and open a new PR.
I got the typings to a state that's decent enough to encourage improvements.